sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Supplier;
+
+import com.fasterxml.classmate.MemberResolver;
+import com.fasterxml.classmate.ResolvedType;
+import com.fasterxml.classmate.ResolvedTypeWithMembers;
+import com.fasterxml.classmate.TypeResolver;
+import com.fasterxml.classmate.members.ResolvedMethod;
+
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.Factory;
+import net.sf.cglib.proxy.MethodInterceptor;
+
+import org.jdbi.v3.Handle;
+import org.jdbi.v3.extension.ExtensionFactory;
+import org.jdbi.v3.sqlobject.mixins.GetHandle;
+import org.jdbi.v3.sqlobject.mixins.Transactional;
+
+public enum SqlObjectFactory implements ExtensionFactory<SqlObject> {
+    INSTANCE;
+
+    private static final MethodInterceptor NO_OP = (proxy, method, args, methodProxy) -> null;
+
+    private final TypeResolver typeResolver = new TypeResolver();
+    private final Map<Method, Handler> mixinHandlers = new HashMap<>();
+    private final ConcurrentMap<Class<?>, Map<Method, Handler>> handlersCache = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Class<?>, Factory> factories = new ConcurrentHashMap<>();
+
+    SqlObjectFactory() {
+        mixinHandlers.putAll(TransactionalHelper.handlers());
+        mixinHandlers.putAll(GetHandleHelper.handlers());
+    }
+
+    @Override
+    public SqlObject createConfig() {
+        return new SqlObject();
+    }
+
+    @Override
+    public boolean accepts(Class<?> extensionType) {
+        if (GetHandle.class.isAssignableFrom(extensionType) ||
+                Transactional.class.isAssignableFrom(extensionType)) {
+            return true;
+        }
+
+        MemberResolver mr = new MemberResolver(typeResolver);
+        ResolvedType extension_type = typeResolver.resolve(extensionType);
+        ResolvedTypeWithMembers d = mr.resolve(extension_type, null, null);
+        for (ResolvedMethod method : d.getMemberMethods()) {
+            Method rawMethod = method.getRawMember();
+            if (rawMethod.isAnnotationPresent(SqlQuery.class)
```

Comment:
Another idea to explore: modify the mixins to use meta-annotations

    public interface GetHandle
    {
      @SqlMethod(GetHandleHandler.class)
      Handle getHandle();
    }

Then the mixins go through the same annotation lookup procedure as e.g. `@SqlQuery`, `@SqlBatch`, etc.